source/common/http/http1/codec_impl.cc

```diff
@@ -1143,13 +1143,38 @@ void ServerConnectionImpl::onBody(Buffer::Instance& data) {
   }
 }
 
+Http::Status ServerConnectionImpl::dispatch(Buffer::Instance& data) {
+  if (active_request_ != nullptr && active_request_->remote_complete_) {
+    // Active downstream request remote complete but there is some new data comming then try to
+    // disable the connection reading. Connection reading will be re-enabled after the current
+    // active downstream request has completed.
+    // This ensures that the new comming data can be consumed after the current active downstream
+    // request has completed.
```

Comment:
This is a bit hard to grok, perhaps:
Read disable the connection if the downstream is sending additional data while we are working on an existing request. Reading from the connection will be re-enabled after the active request is completed.